netlogo-gui/src/main/lab/gui/ProgressDialog.scala

```diff
@@ -213,7 +219,26 @@ private [gui] class ProgressDialog(dialog: java.awt.Dialog, supervisor: Supervis
     }
   }
   override def stepCompleted(w: Workspace, steps: Int) {
-    if (!w.isHeadless) this.steps = steps
+    if (!w.isHeadless) {
+      this.steps = steps
+      if (workspace.triedToExportPlot) {
+        workspace.exportPlotWarningAction match {
+          case ExportPlotWarningAction.Warn => {
+            workspace.setExportPlotWarningAction(ExportPlotWarningAction.Ignore)
+            org.nlogo.awt.EventQueue.invokeLater(new Runnable() {
+              def run() {
+                OptionDialog.showMessage(
+                  workspace.getFrame, "Updating Plots Warning",
+                  "Warning: enable plot updating in Run Options if you want to collect plot data using export-plot, export-all-plots, export-world, or export-interface.",
+                  Array(I18N.gui.get("common.buttons.continue"))
+                )
+              }
+            })
+          }
+          case _ =>
+        }
```

Comment:
Can simplify to the below without a single-case `match`, and also merge the error message with the errors as an I18N string.

```scala
      if (workspace.triedToExportPlot && workspace.exportPlotWarningAction == ExportPlotWarningAction.Warn) {
        workspace.setExportPlotWarningAction(ExportPlotWarningAction.Ignore)
        org.nlogo.awt.EventQueue.invokeLater(new Runnable() {
          def run() {
            OptionDialog.showMessage(
              workspace.getFrame, "Updating Plots Warning",
              "Warning: enable plot updating in Run Options if you want to collect plot data using export-plot, export-all-plots, export-world, or export-interface.",
              Array(I18N.gui.get("common.buttons.continue"))
            )
          }
        })
      }
```

Reply:
Done